src/server/transaction.cc

```diff
@@ -659,7 +661,9 @@ OpStatus Transaction::ScheduleSingleHop(RunnableType cb) {
 
   // If we run only on one shard and conclude, we can avoid scheduling at all
   // and directly dispatch the task to its destination shard.
-  bool schedule_fast = (unique_shard_cnt_ == 1) && !IsGlobal() && !IsAtomicMulti();
+  bool schedule_fast = (unique_shard_cnt_ == 1) && !IsGlobal() && !IsAtomicMulti()
+      // && ServerState::tlocal()->gstate() != GlobalState::TAKEN_OVER
```

Comment:
why did you leave this in comment?

Reply:
I needed it in a previous versions and realized it was redundant, but forgot to delete it. thanks :)